### PR TITLE
Refactor CI Workflow

### PR DESCRIPTION
**Pull request by andistorm**
## Changes
* Split `Build And Test` in multiple jobs
* Provide `dist` and `wheels` artifact
* Install  `ev-cli` in python venv

## Requires
- [x] https://github.com/EVerest/everest-cmake/pull/4
- [x] https://github.com/EVerest/everest-ci/pull/19
- [x] https://github.com/EVerest/everest-utils/pull/146
- [x] https://github.com/EVerest/everest-framework/pull/188
- [x] https://github.com/EVerest/ext-switchev-iso15118/pull/29
- [x] https://github.com/EVerest/EVerest/pull/176
- [x] https://github.com/EVerest/everest-core/pull/622

## TODO before Merging

- [x] Set everest-cmake version
- [x] bump everest-ci version
- [x] bump everest-utils version
- [x] bump everest-framework version
- [x] bump ext-switchev-iso15118 version
- [ ] Update names of required workflows

## TODO after Merging

- [ ] Release/Tag new version

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -17,7 +17,7 @@ if (NOT everest-cmake_FOUND)
     FetchContent_Declare(
         everest-cmake
         GIT_REPOSITORY https://github.com/EVerest/everest-cmake.git
-        GIT_TAG        main
+        GIT_TAG        feature/add-python-cmake-utils
```
Comment (andistorm) [inline CMakeLists.txt:20]:
```suggestion
        GIT_TAG        main
```

**Comment by SirVer** [comment]
Defering review to @golovasteek

**Review by SirVer — APPROVED** [review]

File: .github/workflows/build_and_test.yaml [hunk 2]
```diff
@@ -18,8 +18,22 @@ env:
   BUILD_KIT_IMAGE: ghcr.io/everest/build-kit-alpine:v1.1.4
```
Comment (andistorm) [inline .github/workflows/build_and_test.yaml:18]:
```suggestion
  BUILD_KIT_IMAGE: ghcr.io/everest/build-kit-alpine:v1.2.0
```

**Review by hikinggrass — APPROVED** [review]

**Comment by andistorm** [comment]
Please review @corneliusclaussen or @Pietfried

**Review by Pietfried — APPROVED** [review]

**Pull request merged**